### ISSUE TITLE
Revamp SOCLI

### ISSUE THREAD

**Issue by gautamkrishnar** [gautamkrishnar/socli#149]
As of now this project needs some serious refactoring... There are several pending PRs in this repo that needs to be merged... Also we need to merge several changes from master to develop since most of the previous PRs are merged to the master.

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $20.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/61608885/issues/149) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/61608885/issues/149).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by shivamriky** [comment]
What kind of help do you need on this project?
I found the idea really interesting and would love to spend time if you can provide some guidance.

**Comment by gautamkrishnar** [comment]
Hi @shivamriky please fork the project and work on https://github.com/gautamkrishnar/socli/tree/refactoring (refractoring) branch. I had already merged all prs to that branch... It contains the latest code but it is not functional. Please have a look.

**Comment by shivamriky** [comment]
Great! I will have a look :)

**Comment by lokesh1729** [comment]
@gautamkrishnar any update on this ??? do you need my help???

**Comment by thevirtuoso1973** [comment]
I had a look at your code and ran the tests on refactoring branch. Test 6 failed so I was looking through the implementation of search.py and was confused to the purpose of line 20.
https://github.com/gautamkrishnar/socli/blob/e1a81d2772c7b7c5498b02a826db218b81113755/socli/search.py#L20
Why not just use so_qurl as the default link to query since 'https://stackoverflow.com/?tab=python' (for example) would not provide any results?
https://github.com/gautamkrishnar/socli/blob/e1a81d2772c7b7c5498b02a826db218b81113755/socli/search.py#L41-L42

**Comment by gautamkrishnar** [comment]
@thevirtuoso1973 Thanks for taking it up. Regarding br flag, it was introduced in https://github.com/gautamkrishnar/socli/pull/99/files 

It was meant to be used as a flag for the argument -b that was used to implement browse feature. Please see https://github.com/gautamkrishnar/socli/issues/25 

Now it has gone stale, Please remove that condition checks if its no longer needed.

@lokesh1729 really sorry for missing your comment, i was quite busy. As of now it is almost complete the latest code is on the refactoring branch but it needs a lots of testing before we can release it to the production.

**Comment by isaac-philip** [comment]
is this closed ?
